lib/sensors/inverter_current.cpp

```diff
@@ -0,0 +1,38 @@
+#include <inverter_current.hpp>
+#include "adc_mux.hpp"
+
+namespace hyped::sensors {
+
+std::optional<InverterCurrent> InverterCurrent::create(core::ILogger &logger,
+                                                       std::shared_ptr<io::II2c> i2c,
+                                                       const std::uint8_t adc_mux_channel)
```

Comment:
an idea I just thought of here, change the `kAdcMuxChannelN` values to be an enum where each option has the value of the channel address. then here you have a type guarantee that it will be a valid register.